server/src/test/java/org/elasticsearch/index/mapper/LegacyGeoShapeFieldMapperTests.java

```diff
@@ -98,6 +100,23 @@ protected boolean supportsMeta() {
         return false;
     }
 
+    @Override
+    protected MapperService createMapperService(XContentBuilder mappings) throws IOException {
+        Version version = VersionUtils.randomPreviousCompatibleVersion(random(), Version.V_8_0_0);
+        return createMapperService(version, mappings);
+    }
```

Comment:
How about we also override `createMapperService(Version, XContentBuilder)` and do an assumeFalse that the version is less than Version.V_8_0_0?  It's final in the base class at the moment but I think it's fine to make it overrideable.

Reply:
That works too. Implemented.